crates/dojo-lang/src/print.rs

```diff
@@ -40,3 +40,42 @@ pub fn derive_print(db: &dyn SyntaxGroup, struct_ast: ItemStruct) -> RewriteNode
         ]),
     )
 }
+
+/// Derives PrintTrait for an enum.
+/// Parameters:
+/// * db: The semantic database.
+/// * enum_ast: The AST of the model enum.
+/// Returns:
+/// * A RewriteNode containing the generated code.
+pub fn handle_print_enum(db: &dyn SyntaxGroup, enum_ast: ItemEnum) -> RewriteNode {
+    let enum_name = enum_ast.name(db).text(db);
+    let prints: Vec<_> = enum_ast
+        .variants(db)
+        .elements(db)
+        .iter()
+        .map(|m| {
+            format!(
+                "{}::{}(value) => {{ core::debug::PrintTrait::print('{}'); \
+                 core::debug::PrintTrait::print(value); }}",
+                enum_name,
+                m.name(db).text(db).to_string(),
+                m.name(db).text(db).to_string()
+            )
```

Comment:
Something like this should work, could you please double check and adjust the implementation if needed?
```suggestion
            let variant_name = m.name(db).text(db).to_string();

            match m.type_clause(db) {
                OptionTypeClause::Empty(_) => {
                    format!(
                        "{}::{} => {{ core::debug::PrintTrait::print('{}'); }}",
                        enum_name, variant_name, variant_name,
                    )
                }
                OptionTypeClause::TypeClause(_) => {
                    format!(
                        "{}::{}(value) => {{ core::debug::PrintTrait::print('{}'); \                                                                          
                         core::debug::PrintTrait::print(value); }}",
                        enum_name, variant_name, variant_name,
                    )
                }
            }
```

Reply:
even after doing this it didn't update the test files.

can you let me know how did you figure out that compilation is failing?